static/src/javascripts/projects/commercial/modules/ad-verification/prepare-ad-verification.js

```diff
@@ -8,14 +8,66 @@ const errorHandler = (error: Error) => {
     console.log('Failed to load Confiant:', error);
 };
 
+interface impressionsDfpObject {
+    s: string;  // Slot element ID
+    ad: string; // Advertiser ID
+    c: string;  // Creative ID
+    I: string;  // Line item ID
+    o: string;  // Order ID
+    A: string;  // Ad unit name
+    y: string;  // Yield group ID (Exchange Bidder)
+}
+
+const confiantRefreshedSlots =[];
+
+const refreshBlockedSlotOnce = (
+    blockingType: number,
+    blockingId: string,
+    isBlocked: boolean,
+    wrapperId: string,
+    tagId: string,
+    impressionsData: {
+        preBid: { adId?: string, cpm?: number },
+        dfp: impressionsDfpObject,
+    }
+): Promise<void> => {
+    console.log('*** Already refreshed slots: ', confiantRefreshedSlots);
+    console.log('*** Callback run for slot ', impressionsData.dfp.s);
+    const blockedSlotPath =
+        typeof impressionsData !== 'undefined' &&
+        typeof impressionsData.dfp !== 'undefined'
+            ? impressionsData.dfp.s
+            : null;
+    console.log("*** Has slot already been refreshed? ", confiantRefreshedSlots.includes(blockedSlotPath))
+    // check if ad is blocked and haven't refreshed the slot yet.
+    if (isBlocked && !confiantRefreshedSlots.includes(blockedSlotPath) ) {
+        console.log('*** Ad slot blocked');
+        const slots = window.googletag.pubads().getSlots();
+        slots.forEach((currentSlot) => {
+            if (blockedSlotPath === currentSlot.getSlotElementId()) {
+                console.log('*** Refreshing blocked ad slot ', currentSlot);
+                // refresh the blocked slot to get new ad
+                window.googletag.pubads().refresh([currentSlot]);
```

Comment:
I think this should be calling [refreshAdvert] or [loadAdvert], do in [/dfp/load-advert.js][2]

[refreshAdvert]:https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/commercial/modules/dfp/load-advert.js#L50
[loadAdvert]:https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/commercial/modules/dfp/load-advert.js#L32

[2]: https://github.com/guardian/frontend/tree/master/static/src/javascripts/projects/commercial/modules/dfp/load-advert.js

That is what [displayAds] is calling

[displayAds]: 
https://github.com/guardian/frontend/tree/master/static/src/javascripts/projects/commercial/modules/dfp/display-ads.js#L7